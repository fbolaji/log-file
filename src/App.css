.App {
  text-align: center;
}

.mainTitle {
  background-color: #282c34;
  min-height: 8vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.loader-overlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: black;
  opacity: 0.3;
  z-index: 10004;
}
.loader-circle-wrap {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  height: 100px;
  width: 100px;
  margin: auto;
}
.loader-circle {
  border: 4px solid #ffffff;
  border-top: 4px solid #899091;
  border-radius: 50%;
  width: 100px;
  height: 100px;
  animation: lds-dual-ring 1.2s linear infinite;
}
@keyframes lds-dual-ring {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}