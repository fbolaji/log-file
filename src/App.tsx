import React from 'react';
import './App.css';

import LogsListComponent from "./components/logsList";

function App() {
  return (
    <div className="App">
        <LogsListComponent />
    </div>
  );
}

export default App;
