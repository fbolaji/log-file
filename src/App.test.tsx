import React from "react";
import { render } from '@testing-library/react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from "redux-mock-store";
import App from "./App";
import {Provider} from "react-redux";

configure({ adapter: new Adapter() });

const store = configureStore()({});

describe('App', () => {
  it("Should match snapshot", () => {
    const wrapper = render(<Provider store={store} ><App /></Provider>);
    expect(wrapper).toMatchSnapshot();
  });
});
