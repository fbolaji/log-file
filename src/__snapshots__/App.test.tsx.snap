// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="container-lg"
        >
          <header
            class="mainTitle"
          >
            <h2>
              Asset control logs File
            </h2>
          </header>
          <div
            class="mt-2 row"
          >
            <div
              class="col-md-7 col-12 order-md-1 order-2"
            >
              <div
                class="react-bootstrap-table"
              >
                <table
                  class="table table-striped table-hover table-bordered logsTable"
                >
                  <thead>
                    <tr>
                      <th
                        tabindex="0"
                      >
                        #
                      </th>
                      <th
                        tabindex="0"
                      >
                        DateTime
                      </th>
                      <th
                        tabindex="0"
                      >
                        Severity
                      </th>
                      <th
                        tabindex="0"
                      >
                        Message
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr
                      style="background-color: rgb(52, 58, 64); color: rgb(204, 204, 204);"
                    >
                      <td>
                        1595610796967
                      </td>
                      <td />
                      <td>
                        INFO
                      </td>
                      <td>
                        Some INFO message
                      </td>
                    </tr>
                    <tr
                      style="background-color: rgb(33, 37, 41); color: rgb(204, 204, 204);"
                    >
                      <td>
                        1595610806034
                      </td>
                      <td />
                      <td>
                        WARNING
                      </td>
                      <td>
                        Some WARNING message
                      </td>
                    </tr>
                    <tr
                      style="background-color: rgb(52, 58, 64); color: rgb(204, 204, 204);"
                    >
                      <td>
                        1595610815087
                      </td>
                      <td />
                      <td>
                        INFO
                      </td>
                      <td>
                        Some INFO message
                      </td>
                    </tr>
                    <tr
                      style="background-color: rgb(33, 37, 41); color: rgb(204, 204, 204);"
                    >
                      <td>
                        1595610824112
                      </td>
                      <td />
                      <td>
                        INFO
                      </td>
                      <td>
                        Some INFO message
                      </td>
                    </tr>
                    <tr
                      style="background-color: rgb(52, 58, 64); color: rgb(204, 204, 204);"
                    >
                      <td>
                        1595610833176
                      </td>
                      <td />
                      <td>
                        WARNING
                      </td>
                      <td>
                        Some WARNING message
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <div
                class="row react-bootstrap-table-pagination"
              >
                <div
                  class="col-md-6 col-xs-6 col-sm-6 col-lg-6"
                />
                <div
                  class="react-bootstrap-table-pagination-list col-md-6 col-xs-6 col-sm-6 col-lg-6"
                >
                  <ul
                    class="pagination react-bootstrap-table-page-btns-ul"
                  >
                    <li
                      class="active page-item"
                      title="1"
                    >
                      <a
                        class="page-link"
                        href="#"
                      >
                        1
                      </a>
                    </li>
                    <li
                      class="page-item"
                      title="2"
                    >
                      <a
                        class="page-link"
                        href="#"
                      >
                        2
                      </a>
                    </li>
                    <li
                      class="page-item"
                      title="3"
                    >
                      <a
                        class="page-link"
                        href="#"
                      >
                        3
                      </a>
                    </li>
                    <li
                      class="page-item"
                      title="next page"
                    >
                      <a
                        class="page-link"
                        href="#"
                      >
                        &gt;
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
            <div
              class="col-md-5 col-12 order-md-2 order-1"
            >
              <aside
                data-testid="logStatistic"
              >
                <div>
                  Loading Chart
                </div>
              </aside>
            </div>
          </div>
        </div>
      </div>
    </div>
    <script
      src="https://www.gstatic.com/charts/loader.js"
    />
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="container-lg"
      >
        <header
          class="mainTitle"
        >
          <h2>
            Asset control logs File
          </h2>
        </header>
        <div
          class="mt-2 row"
        >
          <div
            class="col-md-7 col-12 order-md-1 order-2"
          >
            <div
              class="react-bootstrap-table"
            >
              <table
                class="table table-striped table-hover table-bordered logsTable"
              >
                <thead>
                  <tr>
                    <th
                      tabindex="0"
                    >
                      #
                    </th>
                    <th
                      tabindex="0"
                    >
                      DateTime
                    </th>
                    <th
                      tabindex="0"
                    >
                      Severity
                    </th>
                    <th
                      tabindex="0"
                    >
                      Message
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr
                    style="background-color: rgb(52, 58, 64); color: rgb(204, 204, 204);"
                  >
                    <td>
                      1595610796967
                    </td>
                    <td />
                    <td>
                      INFO
                    </td>
                    <td>
                      Some INFO message
                    </td>
                  </tr>
                  <tr
                    style="background-color: rgb(33, 37, 41); color: rgb(204, 204, 204);"
                  >
                    <td>
                      1595610806034
                    </td>
                    <td />
                    <td>
                      WARNING
                    </td>
                    <td>
                      Some WARNING message
                    </td>
                  </tr>
                  <tr
                    style="background-color: rgb(52, 58, 64); color: rgb(204, 204, 204);"
                  >
                    <td>
                      1595610815087
                    </td>
                    <td />
                    <td>
                      INFO
                    </td>
                    <td>
                      Some INFO message
                    </td>
                  </tr>
                  <tr
                    style="background-color: rgb(33, 37, 41); color: rgb(204, 204, 204);"
                  >
                    <td>
                      1595610824112
                    </td>
                    <td />
                    <td>
                      INFO
                    </td>
                    <td>
                      Some INFO message
                    </td>
                  </tr>
                  <tr
                    style="background-color: rgb(52, 58, 64); color: rgb(204, 204, 204);"
                  >
                    <td>
                      1595610833176
                    </td>
                    <td />
                    <td>
                      WARNING
                    </td>
                    <td>
                      Some WARNING message
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div
              class="row react-bootstrap-table-pagination"
            >
              <div
                class="col-md-6 col-xs-6 col-sm-6 col-lg-6"
              />
              <div
                class="react-bootstrap-table-pagination-list col-md-6 col-xs-6 col-sm-6 col-lg-6"
              >
                <ul
                  class="pagination react-bootstrap-table-page-btns-ul"
                >
                  <li
                    class="active page-item"
                    title="1"
                  >
                    <a
                      class="page-link"
                      href="#"
                    >
                      1
                    </a>
                  </li>
                  <li
                    class="page-item"
                    title="2"
                  >
                    <a
                      class="page-link"
                      href="#"
                    >
                      2
                    </a>
                  </li>
                  <li
                    class="page-item"
                    title="3"
                  >
                    <a
                      class="page-link"
                      href="#"
                    >
                      3
                    </a>
                  </li>
                  <li
                    class="page-item"
                    title="next page"
                  >
                    <a
                      class="page-link"
                      href="#"
                    >
                      &gt;
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="col-md-5 col-12 order-md-2 order-1"
          >
            <aside
              data-testid="logStatistic"
            >
              <div>
                Loading Chart
              </div>
            </aside>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
